pkg/config/config.go

```diff
@@ -0,0 +1,302 @@
+	// TODO: Implement reporting configuration
+	// IncludeMetadata    bool `yaml:"include_metadata" json:"include_metadata"`
+	// IncludeRemediation bool `yaml:"include_remediation" json:"include_remediation"`
+	// IncludeReferences  bool `yaml:"include_references" json:"include_references"`
+	// GroupByCategory    bool `yaml:"group_by_category" json:"group_by_category"`
+	// ShowProgress       bool `yaml:"show_progress" json:"show_progress"`
+}
+
+// IntegrationsConfig represents integrations configuration
+// Note: Not yet implemented in the codebase
+type IntegrationsConfig struct {
+	// TODO: Implement integrations configuration
+	// Webhooks []string `yaml:"webhooks" json:"webhooks"`
+	// SARIF    struct {
+	//	IncludeRuleHelp  bool `yaml:"include_rule_help" json:"include_rule_help"`
+	//	IncludeLocations bool `yaml:"include_locations" json:"include_locations"`
+	// } `yaml:"sarif" json:"sarif"`
+}
+
+// DevelopmentConfig represents development configuration
+// Note: Not yet implemented in the codebase
+type DevelopmentConfig struct {
+	// TODO: Implement development configuration
+	// Debug        bool `yaml:"debug" json:"debug"`
+	// DryRun       bool `yaml:"dry_run" json:"dry_run"`
+	// ValidateOnly bool `yaml:"validate_only" json:"validate_only"`
+	// Experimental bool `yaml:"experimental" json:"experimental"`
+}
+
+// SpotterConfig represents the complete Spotter configuration
+type SpotterConfig struct {
+	// Legacy fields for backward compatibility
+	HubURL   string `yaml:"hub_url" json:"hub_url"`
+	APIKey   string `yaml:"api_key" json:"api_key"`
+	CacheDir string `yaml:"cache_dir" json:"cache_dir"`
+	RulesDir string `yaml:"rules_dir" json:"rules_dir"`
+	PacksDir string `yaml:"packs_dir" json:"packs_dir"`
+
+	// Implemented configuration sections
+	Logging    LoggingConfig    `yaml:"logging" json:"logging"`
+	Rules      RulesConfig      `yaml:"rules" json:"rules"`
+	Output     OutputConfig     `yaml:"output" json:"output"`
+	Kubernetes KubernetesConfig `yaml:"kubernetes" json:"kubernetes"`
+	Helm       HelmConfig       `yaml:"helm" json:"helm"`
+
+	// TODO: Uncomment when these features are implemented
+	// Scanner      ScannerConfig      `yaml:"scanner" json:"scanner"`
+	// Filtering    FilteringConfig    `yaml:"filtering" json:"filtering"`
+	// Performance  PerformanceConfig  `yaml:"performance" json:"performance"`
+	// Security     SecurityConfig     `yaml:"security" json:"security"`
+	// Reporting    ReportingConfig    `yaml:"reporting" json:"reporting"`
+	// Integrations IntegrationsConfig `yaml:"integrations" json:"integrations"`
+	// Development  DevelopmentConfig  `yaml:"development" json:"development"`
+}
+
+// DefaultConfig returns the default configuration
+func DefaultConfig() *SpotterConfig {
+	var config SpotterConfig
+	
+	// Parse the embedded default config
+	if err := yaml.Unmarshal(defaultConfigYAML, &config); err != nil {
+		// Fallback to hardcoded defaults if parsing fails
+		homeDir, _ := os.UserHomeDir()
```

Comment:
this shouldn't be handled, it can simply error out if the default config file isn't found, else there could be inconsistencies in the future

Reply:
Fixed, check pelase